src/dlls/mscoree/mscoree.cpp

```diff
@@ -907,9 +907,23 @@ STDAPI GetCORSystemDirectoryInternal(__out_ecount_part_opt(cchBuffer, *pdwLength
     if (pBuffer == NULL)
         IfFailGo(E_POINTER);
 
+#ifdef CROSSGEN_COMPILE
```

Comment:
We don't use the libcoreclr.so here, right? So it would not work without modifications. The LOADInitCoreCLRModules that updates the pal_module calls dlopen on the passed in path to get the module handle and also calls its CoreDllMain.
The crossgen is kind of a strange beast since it is built of static libs that otherwise form the libcoreclr.so. 
It seems that we will rather need to ensure that the PAL_GetPALDirectory would return valid value even in case PAL was initialized using the PAL_Initialize. But for that I'd create a separate github issue.